src/application/job_manager.py

```
@@ -0,0 +1,41 @@
+            self._jobs[job.id] = job
+            return job.id
+
+    async def get_job(self, job_id: str) -> Optional[DownloadJob]:
+        async with self._lock:
+            return self._jobs.get(job_id)
+
+    async def update_job_status(
+        self, 
+        job_id: str, 
+        status: str, 
+        metadata: Optional[Dict[str, Any]] = None
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)